test/e2e/framework.go

```diff
@@ -702,22 +722,85 @@ func (data *TestData) podWaitForRunning(timeout time.Duration, name, namespace s
 	return err
 }
 
-// podWaitForIP polls the K8s apiserver until the specified Pod is in the "running" state (or until
-// the provided timeout expires). The function then returns the IP address assigned to the Pod.
-func (data *TestData) podWaitForIP(timeout time.Duration, name, namespace string) (string, error) {
+// podWaitForIPs polls the K8s apiserver until the specified Pod is in the "running" state (or until
+// the provided timeout expires). The function then returns the IP addresses assigned to the Pod.
+func (data *TestData) podWaitForIPs(timeout time.Duration, name, namespace string) (*PodIPs, error) {
+	return data.podWaitForIPsAndCheckAF(timeout, name, namespace, true)
+}
+
+// hostNetworkPodWaitForIPs polls the K8s apiserver until the specified Pod is in the "running" state (or until
+// the provided timeout expires). The function is called if the Pod is configured with "hostNetwork". The function
+// doesn't require one IP address is assigned in each cluster supported Pod CIDR, for such Pod has only one IP,
+// and uses the Node's internal address directly.
+func (data *TestData) hostNetworkPodWaitForIPs(timeout time.Duration, name, namespace string) (*PodIPs, error) {
```

Comment:
since we have access to the PodSpec, do you think we could just use the `HostNetwork` boolean instead of having 2 different functions?

Reply:
Thanks for reminding, maybe I could read the value of field "Pod.Spec.HostNetwork" in function `podWaitForIPs` directly, then we don't need to add parameters. And we could use a single function for all cases.